include/picongpu/plugins/openPMD/restart/LoadSpecies.hpp

```diff
@@ -70,6 +149,283 @@ namespace picongpu
+                        totalNumParticles,
+                        maxChunkSize,
+                        /* forEachPatch = */
+                        [this, threadParams](
+                            uint64_t loadRound,
+                            uint64_t numParticlesCurrentBatch,
+                            FrameType& mappedFrame,
+                            uint64_t particleLoadOffset)
+                        {
+                            meta::ForEach<
+                                typename NewParticleDescription::ValueTypeSeq,
+                                LoadParticleAttributesFromOpenPMD<boost::mpl::_1>>
+                                loadAttributes;
+                            loadAttributes(
+                                threadParams,
+                                mappedFrame,
+                                particleSpecies,
+                                particleLoadOffset,
+                                numParticlesCurrentBatch);
+
+                            pmacc::particles::operations::splitIntoListOfFrames(
+                                *speciesTmp,
+                                mappedFrame,
+                                numParticlesCurrentBatch,
+                                cellOffsetToTotalDomain,
+                                totalCellIdx_,
+                                *threadParams->cellDescription,
+                                picLog::INPUT_OUTPUT());
+                        });
+                }
+
+                void loadPartialMatches(std::deque<size_t> const& partialMatches, ThreadParams* threadParams) const
+                {
+                    auto [totalNumParticles, maxChunkSize] = numParticlesAndChunkSize(partialMatches);
+
+                    SubGrid<simDim> const& subGrid = Environment<simDim>::get().SubGrid();
+                    pmacc::Selection<simDim> const localDomain = subGrid.getLocalDomain();
+                    pmacc::Selection<simDim> const globalDomain = subGrid.getGlobalDomain();
+                    /* Offset to transform local particle offsets into total offsets for all particles within the
+                     * current local domain.
+                     * @attention A window can be the full simulation domain or the moving window.
+                     */
+                    DataSpace<simDim> localToTotalDomainOffset(localDomain.offset + globalDomain.offset);
+
+                    /* params->localWindowToDomainOffset is in PIConGPU for a restart zero but to stay generic we take
+                     * the variable into account.
+                     */
+                    DataSpace<simDim> const patchTotalOffset
+                        = localToTotalDomainOffset + threadParams->localWindowToDomainOffset;
+                    DataSpace<simDim> const patchExtent = threadParams->window.localDimensions.size;
+                    DataSpace<simDim> const patchUpperCorner = patchTotalOffset + patchExtent;
```

Comment:
Yeah someone please check if all these position computations match up. They seem to do in my tests.